_playwright-tests/tests/API/RepositoryFilteringTest.spec.ts

```diff
@@ -0,0 +1,264 @@
+        'https://non-existent-domain-for-testing-invalid-status.invalid/repo/',
+      ),
+    );
+
+    await test.step('create test repositories', async () => {
+      const repo1 = await repositoriesApi.createRepository({
+        apiRepositoryRequest: {
+          name: `${baseRepoName}-1`,
+          url: 'https://content-services.github.io/fixtures/yum/centirepos/repo02/',
+          distributionArch: 'ppc64le',
+          distributionVersions: ['8'],
+        },
+      });
+      createdRepos.push(repo1);
+
+      const repo2 = await repositoriesApi.createRepository({
+        apiRepositoryRequest: {
+          name: `${baseRepoName}-2`,
+          url: 'https://content-services.github.io/fixtures/yum/centirepos/repo03/',
+          distributionArch: 'x86_64',
+          distributionVersions: ['9'],
+        },
+      });
+      createdRepos.push(repo2);
+
+      const repo3 = await repositoriesApi.createRepository({
+        apiRepositoryRequest: {
+          name: `${baseRepoName}-3`,
+          url: 'https://content-services.github.io/fixtures/yum/centirepos/repo04/',
+          distributionArch: 'x86_64',
+          distributionVersions: ['8'],
+        },
+      });
+      createdRepos.push(repo3);
+
+      expect(createdRepos).toHaveLength(3);
+    });
+
+    await test.step('Wait for repositories to be introspected', async () => {
+      const waitWhilePending = (resp: ApiRepositoryResponse) => resp.status === 'Pending';
+
+      for (const repo of createdRepos) {
+        const getRepository = () =>
+          repositoriesApi.getRepository(<GetRepositoryRequest>{
+            uuid: repo.uuid?.toString(),
+          });
+
+        const updatedRepo = await poll(getRepository, waitWhilePending, 10);
+        expect(updatedRepo.status).toBe('Valid');
+      }
+    });
+
+    await test.step('Test single-value filters', async () => {
+      const x86Response = await repositoriesApi.listRepositories({
+        origin: 'external',
+        arch: 'x86_64',
+      });
+
+      const x86RepoNames = getRepoNames(x86Response);
+      expect(x86RepoNames).toContain(`${baseRepoName}-2`);
+      expect(x86RepoNames).toContain(`${baseRepoName}-3`);
+      expect(x86RepoNames).not.toContain(`${baseRepoName}-1`);
+
+      const version8Response = await repositoriesApi.listRepositories({
+        origin: 'external',
+        version: '8',
+      });
+
+      const version8RepoNames = getRepoNames(version8Response);
+      expect(version8RepoNames).toContain(`${baseRepoName}-1`);
+      expect(version8RepoNames).toContain(`${baseRepoName}-3`);
+      expect(version8RepoNames).not.toContain(`${baseRepoName}-2`);
+
+      const availableVersion9Response = await new RepositoriesApi(client).listRepositories({
+        availableForVersion: '9',
+        origin: 'external',
+      });
+
+      const availableVersion9Names = availableVersion9Response.data?.map((repo) => repo.name) || [];
+      expect(availableVersion9Names).toContain(`${baseRepoName}-2`);
```

Comment:
i'd suggest adding a check to each test to verify the number of repos in the response is equal to the count you expect, so it's clear that the other repos are filtered out